### PR TITLE
Use go-waku's connection status updates

### PR DESCRIPTION
**Pull request by vitvly**
go-waku's connection status updates are pushed from status-go via `wakuv2.peerstats` signal. These updates come in a form of `{IsOnline, HasHistory}`  objects. Hence we now have the ability to fetch wakuv2 peer connectivity status.

### QA notes
Upon connecting to wakuv2 fleets, message sending should be enabled (`Send` button should not be greyed out).

### CODE REVIEW

File: src/status_im/subs.cljs [hunk 1]
```diff
@@ -456,11 +457,13 @@
  :disconnected?
  :<- [:peers-count]
  :<- [:waku/v2-flag]
- (fn [[peers-count wakuv2-flag]]
-   ;; TODO Right now wakuv2 module in status-go
-   ;; does not report peer counts properly,
-   ;; so we always assume that we're connected
-   (if wakuv2-flag false (zero? peers-count))))
+ :<- [:waku/v2-peer-stats]
+ (fn [peers-count wakuv2-flag peer-stats]
```
Comment (flexsurfer) [inline src/status_im/subs.cljs]:
```suggestion
 (fn [[peers-count wakuv2-flag peer-stats]]
```

Comment (vitvly) [reply src/status_im/subs.cljs]: @flexsurfer thanks!! Earlier on I tested it and it worked fine. This mistake slipped in some time ago in the previous PR: https://github.com/status-im/status-react/pull/12197/files#diff-21ed1701d49fc16290b71c7c74bfae9dc355e07910b0d4507e0d29a83aacc3bf. Looking at the code, it is clear why it seemed to "work" :)

**Comment by churik** [comment]
@siphiuel 
offline-online status changing is picked up is application only after relogin on wakuv2 fleets.
**Steps to reproduce:**
- create multiaccount
- switch to wakuv2 (tried on both prod and test)
- turn on airplane mode
- open any chat 
- pay attention to "Send"  button

**Expected result:** shouldn't be available, like on eth.prod
**Actual result:** Send button is available until next login

The same happens if go back from offline to online state - for making Send available relogin is required

**Comment by vitvly** [comment]
Thanks @churik , will check

**Comment by vitvly** [comment]
@churik fixed the online status issue

**Comment by Serhy** [comment]
The https://github.com/status-im/status-react/pull/12376#issuecomment-890921781 issue above - fixed.
However now, after returning back online, - connection to peers and status nodes not established. Thus, `Send` message button is not appears. 

**Steps to reproduce:**
- create multiaccount
- switch to wakuv2 (tried on both prod and test)
- turn on airplane mode
- open any chat 
- pay attention to "Send"  button
- once "Send" button disappears, - turn off airplane mode
- verify connection established fine, 'Send' button appears


**Expected result:** Connected to mailservers, Send button available
**Actual result:** Not connected (after 10+ minutes), not able to send message
Screenshot 2021-09-06 at 16 03 43

**Comment by vitvly** [comment]
@Serhy fixed the issue when going back online.

**Comment by Serhy** [comment]
@siphiuel I tested `12d3744` (https://status-im-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-210913-154944-12d374-pr12376-universal.apk and https://i.diawi.com/9mEUQh) on Android and iOS, however last issue described in https://github.com/status-im/status-react/pull/12376#issuecomment-913634133 still present on both devices: i.e. when returning back to online (from airplane mode) there is no connection to any peers, status nodes are established. 

Same scenario as before: https://github.com/status-im/status-react/pull/12376#issuecomment-913634133
Attaching [Status.log](https://github.com/status-im/status-react/files/7161563/Status.log) and geth.log [geth.log](https://github.com/status-im/status-react/files/7161565/geth.log) here additionally for scenario when airplane mode turned ON and OFF and no way to send message (no connection to peers)

**Comment by vitvly** [comment]
@Serhy fixed the airplane mode issue, tested on my Xiaomi phone. Also added proper peer counts to Connectivity status screen.

**Comment by Serhy** [comment]
@siphiuel I've tested iOS and Android builds `d9c16f0` (https://status-im-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-210921-075555-d9c16f-pr12376-universal.apk and https://i.diawi.com/iuk4bk) - the bug with airplane mode fixed. Turning ON/OFF connection - establishes connection fine and 'Send' button appears. 
But not sure whether it's expected: after switching to waku2.prod/waku2.test fleets, - Chats view doesn't show any chats after adding any chat and endless spinner on Chats view displayed. 
Also, messages are not received between two devices which are online and exchanging messages.
Just in case logs are below:
1) Create new account
2) Switch to `waku2.prod` fleet from `Advanced` -> `Fleets`
3) Join any public chat and send messages
4) Return to Chats view to check the chat is displayed
**Actual result:** 


https://user-images.githubusercontent.com/8749671/134327965-1820bd60-a3e6-47e0-a5cb-a77eb73e9586.mov


[geth.log](https://github.com/status-im/status-react/files/7209613/geth.log)
[Status.log](https://github.com/status-im/status-react/files/7209614/Status.log)

**Comment by vitvly** [comment]
@Serhy thanks, i haven't seen this before. All was good with Chats screen etc. Checking now.

**Comment by vitvly** [comment]
@Serhy fixed, this was related to incorrect status-go merge.

**Comment by Serhy** [comment]
Tested Android builds on Android 11 and Android 9, - and now indeed everything is working fine: airplane mode bug fixed and `Send` button active and working when connected to waku2.prod/waku2.test fleets
iOS build failed the first time but then `go-version.json` conflicted, so iOS didn't check.
@siphiuel could you fix the conflict please, we'll check on iOS and let's merge. 
Autotests failed due to known issues in GH.

**Comment by vitvly** [comment]
@Serhy thanks! Fixed the builds.

**Comment by Serhy** [comment]
All good, everything is working, thanks a lot, @siphiuel

**Comment by vitvly** [comment]
@flexsurfer @rasom @cammellos can you please review again?

File: src/status_im/signals/core.cljs [hunk 2]
```diff
@@ -43,6 +43,13 @@
                           :peers-count peers-count)}
               (mailserver/peers-summary-change previous-summary))))
 
+(fx/defn wakuv2-peer-stats
+  [{:keys [db] :as cofx} peer-stats]
+  (let [previous-stats (:peer-stats db)]
+    (fx/merge cofx
+              {:db (assoc db :peer-stats peer-stats
+                          :peers-count (count (:Peers peer-stats)))})))
```
Comment (flexsurfer) [inline src/status_im/signals/core.cljs]:
```suggestion
    {:db (assoc db :peer-stats peer-stats
                        :peers-count (count (:Peers peer-stats)))}))
```

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/signals/core.cljs (hunk 2 above)
Comment (vitvly) [reply src/status_im/signals/core.cljs]: Thanks, fixed!

**Pull request merged**